phpfarm/docker.sh

```diff
@@ -17,14 +17,23 @@ do
 
     # compile the PHP version
     ./compile.sh $VERSION
+
+    # PHP 5.1 names both CLI and CGI executables 'php'. Rename and move them to
+    # correct dirs. See http://php.net/manual/bg/features.commandline.php
+    if [ "$V" == "5.1" ]; then
+        post='custom/post-install-5.1.6.sh'
+        echo "Running commands from '$post'"
+        /bin/bash "$post" "$VERSION"
+    fi
```

Comment:
This is a bit complicated.

When PHP 5.1 compiles, it outputs both the cli and cgi versions as _php_. See reason below . I think this is a bug as PHP 5.2 onwards doesn't do this.

From [PHP manual](http://php.net/manual/en/features.commandline.introduction.php):
> The name, location and existence of the CLI/CGI binaries will differ depending on how PHP is installed on your system. By default when executing make, both the CGI and CLI are built and placed as sapi/cgi/php-cgi and sapi/cli/php respectively, in your PHP source directory. You will note that both are named php.

Hence the CGI version, which should be named _php-cgi_ but is instead named _php_, overwrites the CLI version in _/phpfarm/inst/php-5.1.6/bin/_ .

As a result the [binary executable detection](https://github.com/fpoirotte/phpfarm/blob/master/src/compile.sh#L342-L353) in _compile.sh_ doesn't detect _php-cgi_, and exits _compile.sh_ altogether, missing the [post-install section](https://github.com/fpoirotte/phpfarm/blob/master/src/compile.sh#L395-L403) .

That was why I had to run the post-install script in _docker.sh_.